code/game/objects/structures/grille.dm

```diff
@@ -54,9 +54,20 @@
 		if(RCD_DECONSTRUCT)
 			return list("mode" = RCD_DECONSTRUCT, "delay" = 20, "cost" = 5)
 		if(RCD_WINDOWGRILLE)
+			var/cost = 8
+			var/delay = 2 SECONDS
+
 			if(the_rcd.window_glass == RCD_WINDOW_REINFORCED)
-				return list("mode" = RCD_WINDOWGRILLE, "delay" = 40, "cost" = 12)
-			return list("mode" = RCD_WINDOWGRILLE, "delay" = 20, "cost" = 8)
+				delay = 4 SECONDS
+				cost = 12
+
+			var/turf/grille_turf = get_turf(src)
+
+			if (grille_turf?.rcd_memory == RCD_MEMORY_WINDOWGRILLE)
+				cost /= RCD_MEMORY_COST_BUFF
+				delay /= RCD_MEMORY_SPEED_BUFF
```

Comment:
wrap these in helper proc to reduce copypaste maybe ?

Reply:
Define might be better, since procs can't return two values without a list.

Reply:
Actually, it could be a pure function of `f(x: RCDResult) -> RCDResult` that gets run, nevermind.